styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -769,10 +769,7 @@ private void deleteShardsForCounter(String counterId) throws IOException {
         .build(), entity -> shards.add(entity.getKey()));
 
     for (List<Key> batch : Lists.partition(shards, MAX_NUMBER_OF_ENTITY_GROUPS_IN_ONE_TRANSACTION)) {
```

Comment:
iirc batch size != `MAX_NUMBER_OF_ENTITY_GROUPS_IN_ONE_TRANSACTION`

Reply:
Yeah, right now it is not a part of transaction.